### PR TITLE
Add docstrings #56

### PR DESCRIPTION
**Pull request by AndKaminer**
Added docstrings. In the future, I'm going to add type hinting, which will make the docstrings more clear. I think a higher level documentation page or two might be helpful, which I will work on after type hinting.

### CODE REVIEW

**Comment by AndKaminer** [comment]
@MarcCote Hopefully this should be good. Thanks!

File: scienceworld/scienceworld.py [hunk 1]
```diff
@@ -120,94 +144,112 @@ def reset(self):
...
 
     # Simplifications
-    def get_simplifications_used(self):
+    def get_simplifications_used(self) -> str:
+        ''' Gets the simplifications being used by the simulator. '''
         return self.server.getSimplificationsUsed()
 
-    def get_possible_simplifications(self):
+    def get_possible_simplifications(self) -> List[str]:
+        '''Gets the 6 possible simplifications. There are 6 simplifictions:
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:152]:
```suggestion
        '''Gets the 6 possible simplifications. Those are:
```

File: scienceworld/scienceworld.py [hunk 2]
```diff
@@ -120,94 +144,112 @@ def reset(self):
...
+            - teleportAction: Teleport action
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:153]:
```suggestion
            - teleportAction: Adds actions to teleport directly to any possible location
```

File: scienceworld/scienceworld.py [hunk 3]
```diff
@@ -120,94 +144,112 @@ def reset(self):
...
+            - selfWateringFlowerPots: Self-watering flower pots
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:154]:
```suggestion
            - selfWateringFlowerPots: Flower pots will water themselves such that the plants won't die
```

File: scienceworld/scienceworld.py [hunk 4]
```diff
@@ -120,94 +144,112 @@ def reset(self):
...
+            - openContainers: Containers open by default
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:155]:
```suggestion
            - openContainers: Containers are open by default
```

File: scienceworld/scienceworld.py [hunk 5]
```diff
@@ -120,94 +144,112 @@ def reset(self):
...
+            - openDoors: Doors open by default
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:156]:
```suggestion
            - openDoors: Doors open are by default
```

File: scienceworld/scienceworld.py [hunk 6]
```diff
@@ -120,94 +144,112 @@ def reset(self):
...
+            - noElectricalAction: Remove the electrical actions
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:157]:
```suggestion
            - noElectricalAction: Remove all `connect X to Y` actions to reduce the action space
```

File: scienceworld/scienceworld.py [hunk 7]
```diff
@@ -120,94 +144,112 @@ def reset(self):
...
+            - easy: use all 5 simplifications
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:158]:
```suggestion
            - easy: use all above simplifications
```

File: scienceworld/scienceworld.py [hunk 8]
```diff
@@ -120,94 +144,112 @@ def reset(self):
...
+        ''' Get the name for the supported tasks in ScienceWorld. '''
         return list(self.server.getTaskNames())
 
-    # Get the maximum number of variations for this task
-    def get_max_variations(self, task_name):
+    def get_max_variations(self, task_name) -> int:
+        ''' Get the maximum number of variations for the tasks. '''
         return self.server.getTaskMaxVariations(infer_task(task_name))
 
     # Get possible actions
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:180]: We can remove all of those comments that were meant to act as docstring.

File: scienceworld/scienceworld.py [hunk 9]
```diff
@@ -261,46 +315,77 @@ def save_run_histories(self, filename_out_prefix):
...
-    def get_random_variation_test(self):
+    def get_random_variation_test(self) -> int:
+        ''' Get a single random variation from those available for the testing set. '''
         return self.server.getRandomVariationTest()
 
     # Gold action sequence
-    def get_gold_action_sequence(self):
+    def get_gold_action_sequence(self) -> List[str]:
+        '''Get the gold action sequence.
+        The gold action sequence is the optimal sequence of actions. This function returns that if it is generated.
```
Comment (MarcCote) [inline scienceworld/scienceworld.py:370]:
```suggestion
        The gold action sequence is a sequence of actions that leads to a winning state (there is no guarantee it is the optimal). This function returns that if it is generated.
```

**Review by MarcCote — CHANGES_REQUESTED** [review]
Great work! I made a couple of change requests but otherwise LGTM.

**Comment by AndKaminer** [comment]
Okay that should be good. Sorry you had to fix my dumb grammar mistakes haha

**Pull request merged**